### PR TITLE
Change structure

### LINKED ISSUE #1536: Improve wazuh-packages directory structure
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.4.x |- | - | - |

Since we are only maintaining one SPEC file on each branch maintaining the old directory structure only makes it harder to maintain the repository.

## Proposed change
```
rpms/
├── build.sh
├── CentOS
│   ├── 5
│   │   ├── i386
│   │   │   ├── CentOS-Base.repo
│   │   │   ├── docker_builder.sh
│   │   │   ├── Dockerfile
│   │   │   └── Vagrantfile
│   │   └── x86_64
│   │       ├── CentOS-Base.repo
│   │       └── Dockerfile
│   ├── 6
│   │   ├── i386
│   │   │   ├── CentOS-Base.repo
│   │   │   └── Dockerfile
│   │   └── x86_64
│   │       ├── CentOS-Base.repo
│   │       └── Dockerfile
│   └── 7
│       ├── aarch64
│       │   └── Dockerfile
│       ├── armv7hl
│       │   ├── build_deps.sh
│       │   └── Dockerfile
│       └── ppc64le
│           └── Dockerfile
├── generate_rpm_package.sh
├── README.md
└── SPECS
    ├── wazuh-agent.spec
    └── wazuh-manager.spec
```

## Tasks
- [ ] Change directory structure
- [ ] Modify docker image building
- [ ] Create and upload images
- [ ] Modify Jenkins pipelines according to the changes

### PR DESCRIPTION
**Pull request by DFolchA**
|Related issue|
|---|
|Closes #1536|

## Description

This PR applies some changes to the directory structure of wazuh packages to improve the workflow in the directory.
## Tests

https://ci.wazuh.info/job/Packages_builder_tier/2115/
https://ci.wazuh.info/job/Packages_builder_tier/2116/
https://ci.wazuh.info/job/Packages_builder_tier/2117/
https://ci.wazuh.info/job/Packages_builder_special/478
https://ci.wazuh.info/job/Packages_builder_special/475

- Build the package in any supported platform
  - [x] Linux
  - [ ] Windows
  - [x] macOS
  - [x] Solaris
  - [x] AIX
  - [ ] HP-UX
- [x] Package installation
- [x] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [ ] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [x] Build the package for x86_64
  - [x] Build the package for i386
  - [x] Build the package for armhf
  - [x] Build the package for aarch64
  - [x] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [x] Build the package for x86_64
  - [x] Build the package for i386
  - [x] Build the package for armhf
  - [x] Build the package for aarch64
  - [x] Package install/remove/install
  - [x] Package install/purge/install
  - [x] Check file permissions after installing the package
- Tests for macOS
  - [x] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [x] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

**Review by okynos — APPROVED** [review]

File: arch/build.sh [hunk 1]
```diff
@@ -77,15 +77,15 @@ if [[ "${future}" == "yes" ]]; then
     if [[ ${specs_path} != ${tmp_specs_path} ]]; then
         cp -r ${specs_path} "${tmp_specs_path}"
     fi
-    cp -r ${specs_path}/${base_version} "${tmp_specs_path}"/${wazuh_version}
+    cp -r "${specs_path}" "${tmp_specs_path}"
```
Comment (alberpilot) [inline arch/build.sh:80]: Why we have the same command?

File: rpms/build.sh [hunk 2]
```diff
@@ -66,8 +66,9 @@ cp -R wazuh-* ${build_dir}/${package_name}
 
 # Including spec file
 if [ "${use_local_specs}" = "no" ]; then
-    curl -sL https://github.com/wazuh/wazuh-packages/tarball/${wazuh_packages_branch} | tar zx
-    specs_path=$(find ./wazuh* -type d -name "SPECS" -path "*rpms*")
+    specs_path="/d_specs"
```
Comment (alberpilot) [inline rpms/build.sh:69]: what is `/d_specs`?

Comment (alberpilot) [reply rpms/build.sh:69]: This variable should have better name

File: rpms/build.sh [hunk 3]
```diff
@@ -82,19 +83,16 @@ if [[ "${future}" == "yes" ]]; then
     old_name="wazuh-${build_target}-${base_version}-${package_release}"
     package_name=wazuh-${build_target}-${wazuh_version}
     old_package_name=wazuh-${build_target}-${base_version}
-    cp -r "${specs_path}/${base_version}" "${specs_path}/${wazuh_version}"
-    cd "${specs_path}/${wazuh_version}"
-    rename "${base_version}" "${wazuh_version}" *${base_version}*
-    cd -
+    cp -r "${specs_path}/" "${specs_path}/"
```
Comment (alberpilot) [inline rpms/build.sh:86]: Please verify if this is necessary.

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: rpms/build.sh [hunk 4]
```diff
@@ -66,8 +66,9 @@ cp -R wazuh-* ${build_dir}/${package_name}
 
 # Including spec file
 if [ "${use_local_specs}" = "no" ]; then
-    curl -sL https://github.com/wazuh/wazuh-packages/tarball/${wazuh_packages_branch} | tar zx
-    specs_path=$(find ./wazuh* -type d -name "SPECS" -path "*rpms*")
+    specs_path="/d_specs"
+    mkdir -p "${specs_path}"
+    curl -L "https://raw.githubusercontent.com/wazuh/wazuh-packages/${wazuh_packages_branch}/rpms/SPECS/wazuh-${build_target}.spec" -o "/d_specs/wazuh-${build_target}.spec"
```
Comment (rauldpm) [inline rpms/build.sh:71]: Use ${specs_path} variable instead of hardcoded value "/d_specs/..."

File: rpms/build.sh (hunk 3 above)
Comment (rauldpm) [reply rpms/build.sh:86]: In fact, this copy command does not make a copy on the same folder?

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, review comments

File: rpms/build.sh (hunk 3 above)
Comment (DFolchA) [reply rpms/build.sh]: Solved [760d801](https://github.com/wazuh/wazuh-packages/pull/1571/commits/760d801a242f59dbd739a73b0bd66b8291aeaf66)

File: rpms/build.sh (hunk 4 above)
Comment (DFolchA) [reply rpms/build.sh]: done [760d801](https://github.com/wazuh/wazuh-packages/pull/1571/commits/760d801a242f59dbd739a73b0bd66b8291aeaf66)

**Pull request merged**